### PR TITLE
[DNM] pacific: mgr/snap_schedule: restart old schedules

### PR DESCRIPTION
**Pull request by mchangir**
backport tracker: https://tracker.ceph.com/issues/55056

---

backport of https://github.com/ceph/ceph/pull/45115
parent tracker: https://tracker.ceph.com/issues/54052

this backport was staged using ceph-backport.sh version 16.0.0.6848
find the latest version at https://github.com/ceph/ceph/blob/master/src/script/ceph-backport.sh

### CODE REVIEW

**Comment by gregsfortytwo** [comment]
Jenkins test this please

**Comment by mchangir** [comment]
jenkins test this please

File: src/pybind/mgr/snap_schedule/fs/schedule_client.py [hunk 1]
```diff
@@ -103,9 +103,22 @@ class SnapSchedClient(CephfsClient):
...
         super(SnapSchedClient, self).__init__(mgr)
-        # TODO maybe iterate over all fs instance in fsmap and load snap dbs?
+        # Each db connection is now guarded by a Lock; this is required to
+        # avoid concurrent DB transactions when more than one paths in a
+        # file-system are scheduled at the same interval eg. 1h; without the
+        # lock, there are races to use the same connection, causing  nested
+        # transactions to be aborted
         self.sqlite_connections = {}
         self.active_timers = {}
+        self.conn_lock = Lock()  # lock to protect add/lookup db connections
```
Comment (nmshelke) [inline src/pybind/mgr/snap_schedule/fs/schedule_client.py:113]: Not seen this `Lock` added in original PR.
Is that part of another backport?

Comment (mchangir) [reply src/pybind/mgr/snap_schedule/fs/schedule_client.py:113]: @nmshelke Here's the link to [line 154 in the original comit on master branch](https://github.com/ceph/ceph/blob/57c2ae079b42cde0e8f3deedc760183af81503b2/src/pybind/mgr/snap_schedule/fs/schedule_client.py#L154)

master branch and pacific have diverged significantly and so this delta that doesn't match with the master branch commit

Comment (lxbsz) [reply src/pybind/mgr/snap_schedule/fs/schedule_client.py:113]: So maybe we should list this difference in the commit comments ?

Comment (mchangir) [reply src/pybind/mgr/snap_schedule/fs/schedule_client.py:113]: actually this PR is supposed to go in after #45189 and hence the diff

**Comment by mchangir** [comment]
@yuriw FYI - https://github.com/ceph/ceph/pull/45671#discussion_r844560845

**Comment by ljflores** [comment]
http://pulpito.front.sepia.ceph.com/yuriw-2022-04-06_14:02:46-rados-wip-yuri4-testing-2022-04-05-1720-pacific-distro-default-smithi/
http://pulpito.front.sepia.ceph.com/lflores-2022-04-07_18:45:23-rados-wip-yuri4-testing-2022-04-05-1720-pacific-distro-default-smithi/

Failures, unrelated:
    https://tracker.ceph.com/issues/53501
    https://tracker.ceph.com/issues/49287
    https://tracker.ceph.com/issues/54071
    https://tracker.ceph.com/issues/54086

There were also some selinux denials in several cephadm tests.

Details:
    1. Exception when running 'rook' task. - Ceph - Orchestrator
    2. podman: setting cgroup config for procHooks process caused: Unit libpod-$hash.scope not found - Ceph - Orchestrator
    3. rados/cephadm/osds: Invalid command: missing required parameter hostname(<string>) - Ceph - Orchestrator
    4. Permission denied when trying to unlink and open /var/log/ntpstats/... - Tools - Teuthology

**Comment by ljflores** [comment]
jenkins test api

**Comment by gregsfortytwo** [comment]
Jenkins test docs

**Comment by yuriw** [comment]
@mchangir @vshankar this has conflict merging with https://github.com/ceph/ceph/pull/45189:


--- pr 45671 --- pulling https://github.com/mchangir/ceph.git branch wip-55056-pacific
remote: Enumerating objects: 20, done.
remote: Counting objects: 100% (15/15), done.
Unpacking objects: 100% (20/20), 7.36 KiB | 538.00 KiB/s, done.
remote: Total 20 (delta 15), reused 15 (delta 15), pack-reused 5
From https://github.com/mchangir/ceph
 * branch                    wip-55056-pacific -> FETCH_HEAD
Auto-merging src/pybind/mgr/snap_schedule/fs/schedule_client.py
CONFLICT (content): Merge conflict in src/pybind/mgr/snap_schedule/fs/schedule_client.py
Auto-merging src/pybind/mgr/snap_schedule/fs/schedule.py
Auto-merging qa/tasks/cephfs/test_snap_schedules.py
CONFLICT (content): Merge conflict in qa/tasks/cephfs/test_snap_schedules.py
Automatic merge failed; fix conflicts and then commit the result.
Unable to access https://github.com/mchangir/ceph.git, retrying..
error: Pulling is not possible because you have unmerged files.
hint: Fix them up in the work tree, and then use 'git add/rm <file>'
hint: as appropriate to mark resolution and make a commit.
fatal: Exiting because of an unresolved conflict.
Traceback (most recent call last):
  File "/wip_master/src/script/build-integration-branch", line 104, in <module>
    raise Exception(message)
Exception: Unable to resolve conflict when merging PR#45671

**Comment by vshankar** [comment]
superseded by https://github.com/ceph/ceph/pull/45906

**Pull request closed without merging**